cargo-miri/bin.rs

```diff
@@ -378,6 +378,74 @@ path = "lib.rs"
     }
 }
 
+enum CargoTargets {
+    All,
+    Filtered { lib: bool, bin: Vec<String>, test: Vec<String> },
+}
+
+impl CargoTargets {
+    fn matches(&self, kind: &str, name: &str) -> bool {
+        match self {
+            CargoTargets::All => true,
+            CargoTargets::Filtered { lib, bin, test } => match kind {
+                "lib" => *lib,
+                "bin" => bin.iter().any(|n| n == name),
+                "test" => test.iter().any(|n| n == name),
+                _ => false,
+            },
+        }
+    }
+}
+
+fn parse_cargo_miri_args(
+    mut args: impl Iterator<Item = String>,
+) -> (CargoTargets, Vec<String>, Vec<String>) {
+    let mut lib_present = false;
+    let mut bin_targets = Vec::new();
+    let mut test_targets = Vec::new();
+    let mut additional_args = Vec::new();
+    while let Some(arg) = args.next() {
+        match arg {
+            arg if arg == "--" => {
+                // Miri arguments begin after the first "--".
+                break;
+            }
+            arg if arg == "--lib" => lib_present = true,
+            arg if arg == "--bin" => {
+                if let Some(binary) = args.next() {
+                    if binary == "--" {
+                        show_error(format!("\"--bin\" takes one argument."));
+                    } else {
+                        bin_targets.push(binary)
+                    }
+                } else {
+                    show_error(format!("\"--bin\" takes one argument."));
+                }
+            }
+            arg if arg.starts_with("--bin=") => bin_targets.push((&arg[6..]).to_string()),
```

Comment:
```suggestion
            arg if arg.starts_with("--bin=") => bin_targets.push((&arg["--bin".len()..]).to_string()),
```
Let's avoid magic constants. Same goes for `--test`.